.github/workflows/pytest.yml

```diff
@@ -2,47 +2,98 @@
 
 name: unit tests
 
-# Controls when the workflow will run
+# Control when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
+  # Trigger the workflow on push or pull request events but only for the main branch
   push:
     branches: [ main ]
   pull_request:
     branches: [ main ]
 
-  # Allows you to run this workflow manually from the Actions tab
+  # Allow this workflow to be run manually from the Actions tab
   workflow_dispatch:
 
 jobs:
-  pytest:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
-
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            label: linux-64
+            prefix: /usr/share/miniconda3/envs/test-env
+          - os: macos-latest
```

Comment:
Does this double the time tests take? I would consider mac compatibility issues to be relatively unimportant, maybe worth a weekly/monthly test.